crates/bevy_transform/src/components/transform.rs

```diff
@@ -1,35 +1,100 @@
-use bevy_math::Mat4;
+use super::{NonUniformScale, Rotation, Scale, Translation};
+use bevy_math::{Mat4, Vec3};
 use bevy_property::Properties;
 use std::fmt;
 
 #[derive(Debug, PartialEq, Clone, Copy, Properties)]
 pub struct Transform {
-    pub value: Mat4,
-    pub sync: bool, // NOTE: this is hopefully a temporary measure to allow setting the transform directly.
-                    // ideally setting the transform automatically propagates back to position / translation / rotation,
-                    // but right now they are always considered the source of truth
+    local: Mat4,
+    global: Mat4,
```

Comment:
It seems like including a local and global Mat4 in all Transforms is a lot of overhead for entities without parents. My intuition is that there should still be two different transform types, but maybe one should just be a newtype of the other.

Reply:
I would caution against two types, as then any systems that work on global transform would not work on local and vice versa. A solution may be to have additional `GlobalTransform` which is never written by the user, but only updated by transform system. Root entities just have their `Transform` copied over and child entities have it calculated from parent. But then again, you introduce redundant data which is not always in sync. I would do benchmarking before going this route.

Reply:
Maybe everything should only have a local transform, and all the global tranform computation should be handled by the rendering system. It could cache all the global transforms in a resouce, and track the cache validity using Mutated.

Reply:
The problem is the global transform is needed in more than one place and eventually in user created code. Having those in an extra cache would mean one more step for the user. So this would become a problem about practicality vs performance.

Reply:
Yeah, that makea sense. I retract my comment.